src/redux/actions/cellSets/runCellSetsClustering.js

```diff
@@ -33,19 +37,22 @@ const runCellSetsClustering = (experimentId, resolution) => async (dispatch, get
     type: CELL_SETS_CLUSTERING_UPDATING,
   });
 
+  const ETag = createObjectHash({
+    experimentId, body, qcPipelineStartDate,
+  });
+
```

Comment:
for the rest of the actions that submit a work request, none of them calls `createObjectHash` and `dispatchWorkRequest`. Why calling it now for this one?

Reply:
Yea, I have to rethink about this, I agree there should be a better way to implement this. This action `dispatchWorkRequest` but does not `fetchWorkResult` because it does not directly fetch the work result directly from S3 and use the returned work result. This action dispatches work request, which, when completed, will tell the browser to reload cell sets.

Arguably, this action indirectly **fetch work result from S3**, via a call to `loadCellSets` in https://github.com/hms-dbmi-cellenics/ui/blob/0bcde96857db3fbb8a8f5798a8a17de8ca6a0002/src/redux/actions/cellSets/updateCellSetsClustering.js#L11. This function gets the `cellsets` bucket, not `work-results`. So, if we use `fetchWork`, this will causes 404 errors, because `fetchWork` contains calls to S3.

Reply:
One option could be to make it so that the worker always sends the work finished notification to the socket for `Etag`, apart from hitting the `experimentUpdates` socket.

In this way we could still call `fetchWorkResult` here so that we get the benefit of the timeout tracking on the client that triggered the clustering while still having clustering broadcast its result.

Reply:
To clarify: With this solution `updateCellSetsClustering` would basically just call `fetchWork` and if that results in error then send the redux error. 

It wouldn't not do anything if the cellSets send the notification for success ,becuase that case is handled in `experimentUpdatesHandler`.

Reply:
Yup sure, that can be done.

Reply:
Done 👍 corresponding worker implementation in https://github.com/hms-dbmi-cellenics/worker/pull/253